ledger-core/virtual/integration/method_test.go

```diff
@@ -348,867 +390,540 @@ func TestVirtual_CallMethodAfterPulseChange(t *testing.T) {
 
-	expectedResult := []byte("finish A.Foo")
+			// add ExecutionMocks to runnerMock
+			{
+				outgoingCall := execution.NewRPCBuilder(outgoingCallRef, objectAGlobal).CallMethod(objectBGlobal, classB, "Foo", []byte("123"))
+				objectAResult := requestresult.New([]byte("finish A.New"), objectAGlobal)
+				objectAResult.SetActivate(reference.Global{}, classA, []byte("state A"))
+				objectAExecutionMock := runnerMock.AddExecutionMock("New")
+				objectAExecutionMock.AddStart(
+					func(ctx execution.Context) {
+						logger.Debug("ExecutionStart [A.New]")
+						require.Equal(t, classA, ctx.Request.Callee)
+						require.Equal(t, outgoingA, ctx.Request.CallOutgoing)
+					},
+					&execution.Update{
+						Type:     execution.OutgoingCall,
+						Error:    nil,
+						Outgoing: outgoingCall,
+					},
+				)
+				objectAExecutionMock.AddContinue(
+					func(result []byte) {
+						logger.Debug("ExecutionContinue [A.New]")
+						require.Equal(t, []byte("finish B.Foo"), result)
+					},
+					&execution.Update{
+						Type:   execution.Done,
+						Result: objectAResult,
+					},
+				)
+
+				runnerMock.AddExecutionMock("Foo").AddStart(
+					func(ctx execution.Context) {
+						logger.Debug("ExecutionStart [B.Foo]")
+						require.Equal(t, objectBGlobal, ctx.Request.Callee)
+						require.Equal(t, objectAGlobal, ctx.Request.Caller)
+						require.Equal(t, []byte("123"), ctx.Request.Arguments)
+					},
+					&execution.Update{
+						Type:   execution.Done,
+						Result: requestresult.New([]byte("finish B.Foo"), objectBGlobal),
+					},
+				)
 
-	outgoingCall := execution.NewRPCBuilder(outgoingCallRef, objectAGlobal).CallMethod(objectAGlobal, class, "Bar", []byte("123")).SetInterference(contract.CallTolerable)
-	objectAExecutionMock := runnerMock.AddExecutionMock("Foo")
-	objectAExecutionMock.AddStart(
-		func(ctx execution.Context) {
-			assert.Equal(t, objectAGlobal, ctx.Object)
-			assert.Equal(t, flags, ctx.Isolation)
-			logger.Debug("ExecutionStart [A.Foo]")
-		},
-		&execution.Update{
-			Type:     execution.OutgoingCall,
-			Error:    nil,
-			Outgoing: outgoingCall,
-		},
-	).AddContinue(func(result []byte) {
-		assert.Equal(t, expectedError, result)
+				runnerMock.AddExecutionClassify("Foo", test.stateB, nil)
+			}
 
-	}, &execution.Update{
-		Type:   execution.Done,
-		Result: requestresult.New(expectedResult, objectAGlobal),
-	})
+			// add checks to typedChecker
+			{
+				typedChecker.VCallRequest.Set(func(request *payload.VCallRequest) bool {
+					assert.Equal(t, objectBGlobal, request.Callee)
+					assert.Equal(t, objectAGlobal, request.Caller)
+					assert.Equal(t, []byte("123"), request.Arguments)
+					assert.Equal(t, payload.CTMethod, request.CallType)
+					assert.Equal(t, uint32(1), request.CallSequence)
+					assert.Equal(t, outgoingCallRef, request.CallReason)
+					assert.Equal(t, callFlags, request.CallFlags)
+					assert.Equal(t, server.GetPulse().PulseNumber, request.CallOutgoing.Pulse())
```

Comment:
Why don't you use here `assertCalleeRequest`?

Reply:
done